src/core/thread/mle_router.cpp

```diff
@@ -3150,6 +3148,10 @@ Error MleRouter::SendChildIdResponse(Child &aChild)
 
     Log(kMessageSend, kTypeChildIdResponse, destination, aChild.GetRloc16());
 
+    // On purpose deferring access to settings after sending the response to comply with
+    // the kMaxChildIdResponseTimeout timing
+    SetChildStateToValid(aChild);
```

Comment:
I am not sure if or how much this can help much the case `Settings` is slow. 

The `SendMessage(*message, destination)` call above will place the prepared message in send-queue (msg won't be sent down to MAC/radio immediately within this call). The send-queue is then serviced from a `Tasklet` later after we return from this method. So I think the call to `Settings` here will still happen before msg is actually sent.

Reply:
You are right, I can reproduce the issue by adding some software delay next to `IgnoreError(mChildTable.StoreChild(aChild));`.

Reply:
If so, is this PR still a valid fix?

There seems no easy fix for this issue. 😢

Reply:
If so, is this PR still a valid fix?

There seems no simple fix though 😭

Reply:
Unfortunately it's not. I'm closing it as can't think of a straight forward solution.